projects/angular/src/forms/datepicker/date-input-base.ts

```diff
@@ -264,33 +175,52 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
     }
   }
 
-  private emitDateOutput(date: Date) {
-    if (!datesAreEqual(date, this.previousDateChange)) {
-      this.dateChange.emit(date);
-      this.previousDateChange = date;
-    } else if (!date && this.previousDateChange) {
-      this.dateChange.emit(null);
-      this.previousDateChange = null;
+  protected datepickerHasFormControl() {
+    return !!this.control;
+  }
+
+  /**
+   * Incase of date range error, both start & end date field valdiation has to be triggered
+   * if either of the field gets updated
+   */
+  protected validateDateRange() {
```

Comment:
nit: I still prefer this to be in the child/ren class/es where this code will apply if possible.

Reply:
To avoid code redundancy its kept in base class, also this method will be triggered only for Date range not for single date selection (already handled conditionally).